regression/contracts/assigns_validity_pointer_04/test.desc

```diff
@@ -1,9 +1,11 @@
-KNOWNBUG
+CORE
 main.c
 --enforce-contract foo --replace-call-with-contract bar --replace-call-with-contract baz _ --pointer-primitive-check
-^EXIT=0$
+^EXIT=10$
 ^SIGNAL=0$
-^VERIFICATION SUCCESSFUL$
+^\[foo.\d+\] line \d+ Check that z is assignable: FAILURE$
+^.* 1 of \d+ failed \(\d+ iteration.*\)$
```

Comment:
Same comment. Why are we checking `1` out of "anything"?

Reply:
Thanks for explaining offline. Resolving this.

Reply:
For this case, it's OK to track `1` of "anything". Pointer primitive check can generate many properties to check and I want to make sure that there is only 1 failure: `Check that z is assignable`.

Reply:
I am a little confused as to why only 1 assertion fails here though.

The assigns clause for `foo` doesn't have `z`, nor does it have `*z`. So shouldn't we fail 2 assertions:
1. the containment check @ assignment `z = malloc...`
2. the subset check @ call `baz()` in the next line (which has `*z` in the assign clause for `baz` but not `foo`)?

Reply:
We discussed this offline. Two important points:
1. There is no containment check here, `malloc` is not treated as a normal function call.
2. There is a bug in checking the subset relation which is not stressed by this test case. We will address it on a separate PR.

Reply:
> There is no containment check here, `malloc` is not treated as a normal function call.

Hmm, but `malloc` just returns a fresh allocation, the left-hand side of the assignment in `z = malloc...`, i.e. the variable `z`, must be checked for containment in the assigns clause.

> There is a bug in checking the subset relation which is not stressed by this test case. We will address it on a separate PR.

On digging a little bit deeper, I realized that the we don't actually need to separately check the assigns clause during replacement. We can go ahead and havoc directly, and those havocs would fail if something is not assignable in the calling context.